### ISSUE TITLE
Call button doesn't appear

### ISSUE THREAD

**Issue by vampik** [iNPUTmice/Conversations#3779]
#### General information

* **Version:** 2.8.6
* **Device:** Samsung Galaxy J5 (2016)
* **Android Version:**  Android 7.1.1 (stock)
* **Server name:** self hosted
* **Server software:** ejabberd 20.04
* **Conversations source:** PlayStore

#### Steps to reproduce

1. Open 1:1 chat with a contact who is using Conversations IM

#### Expected result

Call button should appear

#### Actual result

Call buton doesn't appear. There are only encryption button, file sending button and three dots button.

#### Debug output

````
--------- beginning of main
06-12 17:12:08.310  9031  9031 D conversations: ForegroundService: off
06-12 17:12:08.694  9031  9031 D conversations: initializing database...
06-12 17:12:08.696  9031  9031 D conversations: restoring accounts...
06-12 17:12:08.778  9031  9031 D conversations: restoring conversations...
06-12 17:12:08.801  9031  9031 D conversations: finished restoring conversations in 23ms
06-12 17:12:08.820  9031  9080 D conversations: restoring roster...
06-12 17:12:08.848  9031  9031 D conversations: starting file observer
06-12 17:12:08.931  9031  9031 D conversations: ForegroundService: on
06-12 17:12:08.932  9031  9031 D conversations: update unread count to 0
06-12 17:12:08.996  9031  9080 D conversations: restoring messages...
06-12 17:12:09.125  9031  9031 D conversations: ForegroundService: on
06-12 17:12:09.147  9031  9100 D conversations: start merging phone contacts with roster
06-12 17:12:09.148  9031  9100 D conversations: finished merging phone contacts
06-12 17:12:09.215  9031  9110 D conversations: vamp@vampik.ru: connecting
06-12 17:12:09.238  9031  9031 D conversations: resetting all attempt counts
06-12 17:12:09.329  9031  9080 D conversations: finished restoring messages in 333ms
06-12 17:12:09.996  9031  9109 D conversations: updated dynamic shortcuts
06-12 17:12:12.684  9031  9092 D conversations: found 0 changed files on start up. total=16. (139ms)
06-12 17:12:22.247  9031  9110 D conversations: Resolver: [Result{ip='54.38.159.159', hostame='jabber.vampik.ru', port=5223, directTls=true, authenticated=false, priority=0}, Result{ip='54.38.159.159', hostame='jabber.vampik.ru', port=5222, directTls=false, authenticated=false, priority=0}]
06-12 17:12:22.250  9031  9110 D conversations: verified hostname null
06-12 17:12:22.251  9031  9110 D conversations: vamp@vampik.ru: using values from resolver jabber.vampik.ru/54.38.159.159:5223 tls: true
06-12 17:12:22.629  9031  9110 D conversations: vamp@vampik.ru: protocol=TLSv1.3 cipher=TLS_AES_256_GCM_SHA384
06-12 17:12:22.636  9031  9110 D conversations: vamp@vampik.ru/Conversations.keNL: Authenticating with SCRAM-SHA-1
06-12 17:12:23.019  9031  9110 D conversations: vamp@vampik.ru: logged in
06-12 17:12:23.155  9031  9110 D conversations: vamp@vampik.ru: not counting stanza(IqPacket). Not in smacks session.
06-12 17:12:23.158  9031  9110 D conversations: vamp@vampik.ru: starting service discovery
06-12 17:12:23.184  9031  9110 D conversations: vamp@vampik.ru: server caps came from cache
06-12 17:12:23.222  9031  9110 D conversations: vamp@vampik.ru: stream management(3) enabled (resumable)
06-12 17:12:23.438  9031  9110 D conversations: vamp@vampik.ru: Requesting block list
06-12 17:12:23.512  9031  9110 D conversations: Received blocklist update from server
06-12 17:12:24.149  9031  9110 D conversations: AxolotlService (vamp@vampik.ru): Bundle 1470646814 in PEP was current
06-12 17:12:24.150  9031  9110 D conversations: AxolotlService (vamp@vampik.ru): Announcing device 1470646814
06-12 17:12:24.156  9031  9110 D conversations: vamp@vampik.ru: successfully enabled carbons
06-12 17:12:24.220  9031  9110 D conversations: vamp@vampik.ru: online with resource Conversations.keNL
06-12 17:12:24.238  9031  9100 D conversations: start merging phone contacts with roster
06-12 17:12:24.238  9031  9110 D conversations: vamp@vampik.ru: fetching roster
06-12 17:12:24.239  9031  9100 D conversations: finished merging phone contacts
06-12 17:12:24.266  9031  9109 D conversations: skipping shortcut update
06-12 17:12:24.311  9031  9110 D conversations: vamp@vampik.ru: has push support
06-12 17:12:24.314  9031  9100 D conversations: start merging phone contacts with roster
06-12 17:12:24.315  9031  9100 D conversations: finished merging phone contacts
06-12 17:12:24.326  9031  9110 D conversations: vamp@vampik.ru: running mam query with=*, order=NORMAL, after=1591970410385069, catchup=true, ns=urn:xmpp:mam:2
06-12 17:12:24.326  9031  9109 D conversations: skipping shortcut update
06-12 17:12:24.338  9031  9110 D conversations: vamp@vampik.ru sending csi//inactive
06-12 17:12:24.370  9031  9110 D conversations: vamp@vampik.ru: retrieved own device list: [485977735, 1470646814, 1484691201]
06-12 17:12:24.708  9031  9110 D conversations: vamp@vampik.ru: successfully purged offline messages
06-12 17:12:24.708  9031  9109 D conversations: skipping shortcut update
06-12 17:12:24.790  9031  9448 D conversations: vamp@vampik.ru: persisted roster in 82ms
06-12 17:12:24.811  9031  9100 D conversations: start merging phone contacts with roster
06-12 17:12:24.812  9031  9100 D conversations: finished merging phone contacts
06-12 17:12:24.820  9031  9109 D conversations: skipping shortcut update
06-12 17:12:25.509  9031  9110 D conversations: vamp@vampik.ru: making disco request for KWQQsqlerqQGIyKLIQTBsn7Adho= to marina@messag.i.ng/Conversations.rmKw
06-12 17:12:25.678  9031  9110 D conversations: vamp@vampik.ru: making disco request for zcIke+Rk13ah4d1pwDG7bEZsVwA= to tvilla@messag.i.ng/Conversations.YgWY
06-12 17:12:26.201  9031  9110 D conversations: vamp@vampik.ru: finished mam after 0(0) messages. messages left=true count=538
06-12 17:12:26.201  9031  9110 D conversations: vamp@vampik.ru: found 0 pending receipt requests
06-12 17:12:26.344  9031  9110 D conversations: AxolotlService (vamp@vampik.ru): Received PEP device list [485977735, 1470646814, 1484691201] update from vamp@vampik.ru, processing...
06-12 17:12:26.344  9031  9110 D conversations: vamp@vampik.ru: ignoring duplicate own device id list
06-12 17:12:26.384  9031  9110 D conversations: vamp@vampik.ru: 0 bookmarks have been removed
06-12 17:12:26.385  9031  9110 D conversations: vamp@vampik.ru: processing bookmark PEP event
06-12 17:12:26.409  9031  9110 D conversations: vamp@vampik.ru: successfully enabled push on server
06-12 17:12:27.512  9031  9100 D conversations: start merging phone contacts with roster
06-12 17:12:27.513  9031  9100 D conversations: finished merging phone contacts
06-12 17:12:27.521  9031  9109 D conversations: skipping shortcut update
--------- beginning of system
06-12 17:12:36.209  9031  9094 D conversations: deleted file /storage/emulated/0/Android/data/com.google.android.youtube/cache/cache1660803736probe internal=false, database hits=0
06-12 17:12:56.032  9031  9031 D conversations: vamp@vampik.ru send ping (action=eu.siacs.conversations.POST_CONNECTIVITY_CHANGE,lowTimeout=false)
06-12 17:13:01.237  9031  9094 D conversations: deleted file /storage/emulated/0/Android/data/org.telegram.messenger/cache/000000000_999999_temp.jpg internal=false, database hits=0
06-12 17:13:01.254  9031  9094 D conversations: deleted file /storage/emulated/0/Telegram/Telegram Images/000000000_999999.jpg internal=false, database hits=0
06-12 17:13:01.301  9031  9094 D conversations: deleted file /storage/emulated/0/Android/data/org.telegram.messenger/cache/000000000_999999_temp.mp4 internal=false, database hits=0
06-12 17:13:01.307  9031  9094 D conversations: deleted file /storage/emulated/0/Telegram/Telegram Video/000000000_999999.mp4 internal=false, database hits=0
06-12 17:13:01.328  9031  9094 D conversations: deleted file /storage/emulated/0/Android/data/org.telegram.messenger/cache/000000000_999999_temp.ogg internal=false, database hits=0
06-12 17:13:01.335  9031  9094 D conversations: deleted file /storage/emulated/0/Telegram/Telegram Audio/000000000_999999.ogg internal=false, database hits=0
06-12 17:13:01.350  9031  9094 D conversations: deleted file /storage/emulated/0/Android/data/org.telegram.messenger/cache/000000000_999999_temp.doc internal=false, database hits=0
06-12 17:13:01.361  9031  9094 D conversations: deleted file /storage/emulated/0/Telegram/Telegram Documents/000000000_999999.doc internal=false, database hits=0
06-12 17:19:00.034  9031  9031 D conversations: vamp@vampik.ru send ping (action=ping,lowTimeout=false)
06-12 17:23:58.460  9031  9031 D conversations: action: null
06-12 17:23:58.509  9031  9031 D conversations: ConversationsOverviewFragment.onStart()
06-12 17:23:58.512  9031  9031 D conversations: ConversationsOverviewFragment.onResume()
06-12 17:23:58.698  9031  9031 D conversations: vamp@vampik.ru send ping (action=ui,lowTimeout=false)
06-12 17:23:58.826  9031  9031 D conversations: app switched into foreground
06-12 17:23:58.878  9031  9110 D conversations: AxolotlService (vamp@vampik.ru): Received PEP device list [1638024917, 330914352, 1398007000] update from marina@messag.i.ng, processing...
06-12 17:23:58.902  9031  9110 D conversations: AxolotlService (vamp@vampik.ru): Received PEP device list [1677387234, 1597323084] update from tvilla@messag.i.ng, processing...
06-12 17:23:59.128  9031  9031 D conversations: EmojiService succeeded in loading fonts
06-12 17:23:59.160  9031  9031 D conversations: ConversationsOverviewFragment.onPause()
06-12 17:24:01.231  9031  9031 D conversations: ConversationsOverviewFragment.onResume()
06-12 17:24:02.709  9031  9031 D conversations: ConversationFragment.onAttach()
06-12 17:24:02.709  9031  9031 D conversations: ConversationsOverviewFragment.onPause()
06-12 17:24:02.774  9031  9031 D conversations: reInit(hasExtras=true)
06-12 17:24:02.774  9031  9031 D conversations: trimming conversation
06-12 17:24:02.804  9031  9031 D conversations: scrolledToBottomAndNoPending=true
06-12 17:24:02.804  9031  9031 D conversations: jump to first unread message
06-12 17:24:02.982  9031  9031 D conversations: load more messages for Любовь Тихомирова prior to 2020-05-11T08:18:02.114Z
06-12 17:24:02.991  9031  9080 D conversations: vamp@vampik.ru: running mam query with=tvilla@messag.i.ng, end=2020-05-11T08:18:02.114Z, order=REVERSE, catchup=false, ns=urn:xmpp:mam:2
06-12 17:24:03.108  9031  9110 D conversations: skipping duplicate message with tvilla@messag.i.ng/Conversations.V3km. serverMsgIdUpdated=false
06-12 17:24:03.111  9031  9110 D conversations: vamp@vampik.ru: finished mam after 1(0) messages. messages left=false count=1
06-12 17:24:03.111  9031  9110 D conversations: vamp@vampik.ru: found 0 pending receipt requests
06-12 17:24:03.202  9031  9031 D conversations: load more messages for Любовь Тихомирова prior to 2020-05-11T08:18:02.114Z
06-12 17:24:05.967  9031  9031 D conversations: ConversationsOverviewFragment.onStart()
06-12 17:24:05.967  9031  9031 D conversations: ConversationsOverviewFragment.onResume()
06-12 17:24:06.782  9031  9031 D conversations: ConversationFragment.onAttach()
06-12 17:24:06.782  9031  9031 D conversations: ConversationsOverviewFragment.onPause()
06-12 17:24:06.804  9031  9031 D conversations: reInit(hasExtras=true)
06-12 17:24:06.804  9031  9031 D conversations: trimming conversation
06-12 17:24:06.840  9031  9031 D conversations: scrolledToBottomAndNoPending=true
06-12 17:24:06.840  9031  9031 D conversations: jump to first unread message
06-12 17:24:12.409  9031  9031 D conversations: ConversationsOverviewFragment.onStart()
06-12 17:24:12.409  9031  9031 D conversations: ConversationsOverviewFragment.onResume()
06-12 17:24:42.645  9031  9031 D conversations: ConversationsOverviewFragment.onPause()
06-12 17:24:42.737  9031  9031 D conversations: app switched into background

````

My main device is Motorola G7 Plus (Android 9). Call button appears there.
I've installed Conversations IM on the second device (Samsung Galaxy J5 (2016), Android 7.1.1) and call button doesn't appear here. It worth mentioning that I can receive calls. But I cannot make them because there is no button.

**Comment by vampik** [comment]
Both devices use the same account btw

**Comment by licaon-kter** [comment]
Same server you mean?

**Comment by vampik** [comment]
Same server and same JID.

**Comment by licaon-kter** [comment]
You connect to yourself and want to call yourself?

**Comment by iNPUTmice** [comment]
Here is a list of requirements for the call button to appear: https://gist.github.com/iNPUTmice/a28c438d9bbf3f4a3d4c663ffaa224d9#notes-for-users

**Comment by vampik** [comment]
> 
> 
> You connect to yourself and want to call yourself?

No, I don't want to call myself.

I have two phones with Conversation IM, configured with the same account.

The first phone is able to make and receive calls to **another** contacts.
The second phone is able only to receive calls as the call button doesn't appear.

**Comment by licaon-kter** [comment]
@vampik disconnect and reconnect? Restart device?

**Comment by vampik** [comment]
> Here is a list of requirements for the call button to appear: 

Me and my contact use Conversations 2.8.6
We have each other in our contact lists.
The contact is online.

The problem is that on one phone the call button does appear, but it doesn't appear on the second phone. Both phones use the same account.

**Comment by vampik** [comment]
@iNPUTmice May be you misunderstood me.

Let's say my JID is bob@example.org.
I have two phones. Both are setup to use this JID, Let's call them Bob #1 and Bob #2.

Now I want talk with alice@example.org. Let's call this contact Alice.

Bob #1 is able to place a call to Alice (there is a call button), but Bob #2 is not able to call Alice (no call button).

When Alice calls Bob, both phones are ringing simultaneously and are able to receive calls.

**Comment by vampik** [comment]
> 
> 
> @vampik disconnect and reconnect? Restart device?

Tried this. Also tried to clear the application data and setup again with the same result.

**Comment by vampik** [comment]
Here are the example screenshots. Two devices, same account, same contact I want to call to, same time, even the network.

Device #1 has the call button. Device #2 doesn't have the call button.

![Screenshot_20200612-191431](https://user-images.githubusercontent.com/726409/84523864-b73bf380-ace1-11ea-82fd-35cfc581080e.png)

![Screenshot_20200612-191535](https://user-images.githubusercontent.com/726409/84523886-c0c55b80-ace1-11ea-9a67-eb8ee972ff28.jpg)

**Comment by licaon-kter** [comment]
What (type of) VPN is that?

**Comment by vampik** [comment]
> What (type of) VPN is that?

This is Shadowsocks. It is used only for the web browser. All other applications are directly connected to the Internet. Disabling it doesn't change anything.

**Comment by vampik** [comment]
I'm able to do any possible debugging (device logs, server logs, tcpdump, appdata contents as both devices are rooted) - just tell me what to do.

**Comment by iNPUTmice** [comment]
Deactivate (disconnect) your account on the broken device. Start `adb logcat -s conversations`. Activate the account. Wait until it is connected. Toggle air plane mode. Wait until account reconnected. Stop adb logcat.

**Comment by iNPUTmice** [comment]
You seem to have multiple contacts with Conversations. Do none of them show the call button?

**Comment by vampik** [comment]
> Activate the account. Wait until it is connected. Toggle air plane mode. Wait until account reconnected. Stop adb logcat.

[logcat.txt](https://github.com/iNPUTmice/Conversations/files/4772272/logcat.txt)

**Comment by vampik** [comment]
> 
> 
> You seem to have multiple contacts with Conversations. Do none of them show the call button?

Yes, none of them show the call button on the "broken" device.

**Comment by iNPUTmice** [comment]
We are talking about the contacts marina and tvilla here, right?

Can you exchange messages with them?

The reason I’m asking is that here
```
06-12 20:13:51.361 25372 27325 D conversations: vamp@vampik.ru: making disco request for KWQQsqlerqQGIyKLIQTBsn7Adho= to marina@messag.i.ng/Conversations.rmKw
06-12 20:13:51.679 25372 27325 D conversations: vamp@vampik.ru: making disco request for zcIke+Rk13ah4d1pwDG7bEZsVwA= to tvilla@messag.i.ng/Conversations.YgWY
```
we are making a request to discover what features they support. (Asking their devices)

By the time of the reconnect:
```
06-12 20:15:23.027 25372 28471 D conversations: vamp@vampik.ru: SM catchup complete (messages=0, pending IQs=2)
```
we apparently have not yet received an answer. (pending requests 2). Strictly speaking those pending requests doesn’t have to be the two discovery requests; but seeing that the account doesn’t do much else this is likely.

**Comment by vampik** [comment]
> We are talking about the contacts marina and tvilla here, right?

Right.

> Can you exchange messages with them?

I exchanged messages with one of the contacts, but it didn't help. After that tried to reconnect and even restarted the device but it didn't help too.

But after that I decided to restart ejabberd.

I don't know what have helped: the restart itself or the fact that it forced all contacts to reconnect. But the call button appeared on the both contacts!

Could it be related to the fact that my server has XEP-0357: Push Notifications enabled for battery saving?
